### PR TITLE
AccountId instead of address for transaction backend

### PR DESCRIPTION
**Pull request by supermassive**
Resolves https://github.com/brave/brave-browser/issues/31983

- Replacing string address with mojom::AccountId in TxMeta and related code.
- No explicit migration. Just try find account by unique account key, or fallback to legacy address.
- Removed 'from' field from FilTxData as it is a duplicate of TxMeta's from field. 
- address field will be removed from mojom::TransactionInfo in the follow up PR.

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by wchen342 — APPROVED** [review]
Android test++

File: components/brave_wallet/browser/keyring_service.cc [hunk 1]
```diff
@@ -2564,18 +2587,25 @@ KeyringService::SignMessageByBitcoinKeyring(
...
-std::vector<mojom::AccountInfoPtr> KeyringService::GetAllAccountInfos() {
-  std::vector<mojom::AccountInfoPtr> account_infos;
-  for (const auto& keyring_id : GetSupportedKeyrings()) {
-    for (auto& account_info : GetAccountInfosForKeyring(keyring_id)) {
-      account_infos.push_back(std::move(account_info));
+void KeyringService::ResetAllAccountInfosCache() {
+  account_info_cache_.reset();
+}
+
+const std::vector<mojom::AccountInfoPtr>& KeyringService::GetAllAccountInfos() {
```
Comment (StephenHeaps) [inline components/brave_wallet/browser/keyring_service.cc:2594]: On fresh install/profile I'm seeing a nil `selected_account` returned from `GetAllAccountInfos()` when restoring a wallet on iOS on this branch. Creating a wallet always has non-nil `selected_account`; seems specific to restoring wallet on fresh install/profle.

Comment (supermassive) [reply components/brave_wallet/browser/keyring_service.cc:2585]: Couldn't reproduce anything similar with my desktop setup and this PR doesn't touch account selections. Maybe there is some iOS-specific case. Can we debug it separately? Does it block this PR?

File: components/brave_wallet/browser/account_resolver_delegate_impl.cc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "brave/components/brave_wallet/browser/account_resolver_delegate_impl.h"
+
+namespace brave_wallet {
+
+AccountResolverDelegateImpl::AccountResolverDelegateImpl(
+    KeyringService* keyring_service)
+    : keyring_service_(keyring_service) {}
+
+mojom::AccountIdPtr AccountResolverDelegateImpl::ResolveAccountId(
```
[the remark this answers is not in the document]
Comment (supermassive) [reply components/brave_wallet/browser/account_resolver_delegate_impl.cc:14]: Added https://github.com/brave/brave-core/pull/19435/commits/9891a561d25ab2a563ed11e0c5aee81be90b2900

File: components/brave_wallet/browser/account_resolver_delegate.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+#include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
+
+namespace brave_wallet {
+
+class AccountResolverDelegate {
+ public:
+  virtual ~AccountResolverDelegate() = default;
+
+  virtual mojom::AccountIdPtr ResolveAccountId(
```
[the remark this answers is not in the document]
Comment (supermassive) [reply components/brave_wallet/browser/account_resolver_delegate.h:21]: https://github.com/brave/brave-core/commit/9891a561d25ab2a563ed11e0c5aee81be90b2900

File: components/brave_wallet/browser/fil_tx_manager.cc [hunk 4]
```diff
@@ -109,19 +113,12 @@ void FilTxManager::ContinueAddUnapprovedTransaction(
 void FilTxManager::AddUnapprovedTransaction(
     const std::string& chain_id,
     mojom::TxDataUnionPtr tx_data_union,
-    const std::string& from,
+    const mojom::AccountIdPtr& from,
     const absl::optional<url::Origin>& origin,
     const absl::optional<std::string>& group_id,
     AddUnapprovedTransactionCallback callback) {
   DCHECK(tx_data_union->is_fil_tx_data());
-  if (!FilAddress::IsValidAddress(from)) {
```
[the remark this answers is not in the document]
Comment (supermassive) [reply components/brave_wallet/browser/fil_tx_manager.cc:117]: `TxService::AddUnapprovedTransaction` now tries to resolve account id by address which when found implies correct fil address.

File: components/brave_wallet/browser/keyring_service.cc [hunk 5]
```diff
@@ -1176,6 +1182,7 @@ mojom::AccountInfoPtr KeyringService::AddAccountSync(
   SetSelectedAccountInternal(*account);
   NotifyAccountsAdded(*account);
 
+  ResetAllAccountInfosCache();
```
[the remark this answers is not in the document]
Comment (supermassive) [reply components/brave_wallet/browser/keyring_service.cc]: https://github.com/brave/brave-core/commit/9891a561d25ab2a563ed11e0c5aee81be90b2900
That was my initial way of implementing this but didn't work without reordering of some calls around `NotifyAccountsChanged`. Now after some thinking I think reordering is OK

File: components/brave_wallet/browser/keyring_service.h [hunk 6]
```diff
@@ -226,7 +230,8 @@ class KeyringService : public KeyedService, public mojom::KeyringService {
       const mojom::BitcoinKeyId& key_id,
       base::span<const uint8_t, 32> message);
 
-  std::vector<mojom::AccountInfoPtr> GetAllAccountInfos();
+  void ResetAllAccountInfosCache();
```
[the remark this answers is not in the document]
Comment (supermassive) [reply components/brave_wallet/browser/keyring_service.h]: https://github.com/brave/brave-core/commit/9891a561d25ab2a563ed11e0c5aee81be90b2900

**Review by darkdh — APPROVED** [review]

**Review by onyb — APPROVED** [review]
++

**Pull request merged**